lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.h

```diff
@@ -517,6 +517,19 @@ class SwiftLanguageRuntime : public LanguageRuntime {
                              unsigned num_indirections = 0);
 };
 
+/// The target specific register numbers used for async unwinding.
+///
+/// For UnwindPlans, these use eh_frame / dwarf register numbering.
+struct AsyncUnwindRegisterNumbers {
+  uint32_t async_ctx_regnum;
+  uint32_t fp_regnum;
+  uint32_t pc_regnum;
+
+  lldb::RegisterKind GetRegisterKind() const { return lldb::eRegisterKindDWARF; }
```

Comment:
This could use a doxygen comment, it's unclear what the other options would be?

Reply:
Any of the kinds in `lldb::RegisterKind` :)  LLDB has many different numbering schemes for registers, @jasonmolenda calls them "register domains", so whenever a register number is used, it is important to know which domain it is part of. One day I'd like to see if we can design a different _type_ for each domain.

```
enum RegisterKind {
  eRegisterKindEHFrame = 0, ///< the register numbers seen in eh_frame
  eRegisterKindDWARF,       ///< the register numbers seen DWARF
  eRegisterKindGeneric, ///< insn ptr reg, stack ptr reg, etc not specific to
                        ///< any particular target
  eRegisterKindProcessPlugin, ///< num used by the process plugin - e.g. by the
                              ///< remote gdb-protocol stub program
  eRegisterKindLLDB,          ///< lldb's internal register numbers
  kNumRegisterKinds
```